From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 28 Sep 2019 19:11:45 +0200
Subject: [PATCH] {MAYBE UPS} drivers/net/ethernet/mellanox/mlx5: avoid
 emitting __umoddi3

gcc on x86-32 emits a call to __umoddi3 when a u64 is moduled with a u32
value. This function is not implemented there.
---
 drivers/net/ethernet/mellanox/mlx5/core/steering/dr_icm_pool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/steering/dr_icm_pool.c b/drivers/net/ethernet/mellanox/mlx5/core/steering/dr_icm_pool.c
index 913f1e5aaaf2..309cf421e185 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/steering/dr_icm_pool.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/steering/dr_icm_pool.c
@@ -137,7 +137,7 @@ dr_icm_pool_mr_create(struct mlx5dr_icm_pool *pool,
 
 	icm_mr->icm_start_addr = icm_mr->dm.addr;
 
-	align_diff = icm_mr->icm_start_addr % align_base;
+	align_diff = ((uintptr_t)icm_mr->icm_start_addr) % align_base; /* Avoid emitting __umoddi3 for u64 % u32 */
 	if (align_diff)
 		icm_mr->used_length = align_base - align_diff;
 
-- 
